[docs] publishing local build plugin (#71)
expo/eas-build: DEVELOPMENT.md

#### Introducing breaking changes for job API (@expo/eas-build-job package)

> If you want to introduce breaking changes to the `@expo/eas-build-job` package contact one of the [CODEOWNERS](/CODEOWNERS), so we can make sure that legacy cases are still supported on EAS build servers and in GraphQL API. Please describe what changes you want to make and why and we will implement api changes that support both cases. After everything is deployed to production you can introduce PR that relies on the new implementation.

Reviewer: ```suggestion
If you want to introduce breaking changes to the `@expo/eas-build-job` package contact one of the [CODEOWNERS](/CODEOWNERS), so we can make sure that legacy cases are still supported on EAS build servers and in GraphQL API. Please describe what changes you want to make and why and we will implement API changes that support both cases. After everything is deployed to production you can introduce PR that relies on the new implementation.
```

> If you want to introduce breaking changes to the `@expo/eas-build-job` package contact one of the [CODEOWNERS](/CODEOWNERS), so we can make sure that legacy cases are still supported on EAS build servers and in GraphQL API. Please describe what changes you want to make and why and we will implement API changes that support both cases. After everything is deployed to production you can introduce PR that relies on the new implementation.

Reviewer 2: ```suggestion
If you want to introduce breaking changes to the `@expo/eas-build-job` package contact one of the [CODEOWNERS](/CODEOWNERS), so we can make sure that legacy cases are still supported on EAS build servers and in the GraphQL API. Please describe what changes you want to make and why and we will implement API changes that support both cases. After everything is deployed to production you can introduce a PR that relies on the new implementation.
```

> Change like that could be e.g. introducing a new required field in the job object. In that case, we would make changes to allow this field and provide default values for legacy requests, but it would be ignored by the build process. After everything is deployed/published, your follow-up PR would introduce logic that is using this new field.

Reviewer 2: ```suggestion
A breaking change like that could be introducing a new required field on the job object. In that case, we would make changes to allow this field and provide default values for legacy requests, but it would be ignored by the build process. After everything is deployed/published, your follow-up PR would introduce logic that is using this new field.
```

> ### Updating `local-build-plugin`

Reviewer: ```suggestion
### Updating `local-build-plugin` in EAS CLI
```

> - Run `./scripts/publish.js` (local build plugin will be published as `next`).

Reviewer 2: ```suggestion
- Run `./scripts/publish.js` (`local-build-plugin` will be published as `next`).
```

> - If there are breaking changes update `eas-build-job` package in `eas-cli` and make sure everything works.

Reviewer: ```suggestion
- If there are breaking changes, update `eas-build-job` package in EAS CLI and make sure everything works as expected.
```

> - If there are breaking changes, update `eas-build-job` package in EAS CLI and make sure everything works as expected.

Reviewer 2: ```suggestion
- If there are breaking changes, update the `eas-build-job` package in EAS CLI and make sure everything works as expected.
```

> - Tag newly published version as `latest`(`npm dist-tag add eas-cli-local-build-plugin@{version} latest`).

Reviewer: ```suggestion
- Tag newly published version as `latest` - `npm dist-tag add eas-cli-local-build-plugin@VERSION latest`.
```

> - Publish new eas-cli version. `eas-cli` is using fixed version of the `eas-cli-local-build-plugin` package, version is set based on `latest` tag at the time of publishing.

Reviewer: ```suggestion
- Publish a new EAS CLI version. EAS CLI uses a fixed version of the `eas-cli-local-build-plugin` package, the version is set based on `latest` tag at the publish time.
```
